### ISSUE TITLE
MethodHasTooManyParameters false positive on main block of a class caused by event parameters

### ISSUE THREAD

**Issue by ccecvb** [Riverside-Software/sonar-openedge#1136]
I activate the rule yesterday, behaviour could be present for longer.

Main block has 18 parameters, which is greater than 8 authorized

![image](https://github.com/user-attachments/assets/6bb69857-7e15-41ab-b298-2209728286a6)
At first, I thought it was the usings causing this. 

It turns out it's the event parameters that are counted as main block parameters.

```
/*------------------------------------------------------------------------
    File        : ToomanyParams
  ----------------------------------------------------------------------*/
block-level on error undo, throw.

class ToomanyParams:

    define public event Event1 signature void ( input sender as Progress.Lang.Object, input e as Progress.Lang.Object ).
    define public event Event2 signature void ( input sender as Progress.Lang.Object, input e as Progress.Lang.Object ).
    define public event Event3 signature void ( input sender as Progress.Lang.Object, input e as Progress.Lang.Object ).
    define public event Event4 signature void ( input sender as Progress.Lang.Object, input e as Progress.Lang.Object ).
    define public event Event5 signature void ( input sender as Progress.Lang.Object, input e as Progress.Lang.Object ).
    define public event Event6 signature void ( input sender as Progress.Lang.Object, input e as Progress.Lang.Object ).
    define public event Event7 signature void ( input sender as Progress.Lang.Object, input e as Progress.Lang.Object ).
    define public event Event8 signature void ( input sender as Progress.Lang.Object, input e as Progress.Lang.Object ).
    define public event Event9 signature void ( input sender as Progress.Lang.Object, input e as Progress.Lang.Object ).

end class.

```

**Comment by gquerret** [comment]
Confirmed, the parameters of the event are incorrectly attached to the main scope.

**Comment by gquerret** [comment]
Fixed in develop branch
